chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/edition2015/InvalidCriteriaCombinationReviewer.java

```diff
@@ -1,68 +1,66 @@
-
-    @Value("${criterion.170_315_g_10}")
-    private Integer criteriaG10Id;
-
-    private ErrorMessageUtil msgUtil;
-    private FF4j ff4j;
-
+public class InvalidCriteriaCombinationReviewer extends InvalidCriteriaCombination implements Reviewer {
     @Autowired
     public InvalidCriteriaCombinationReviewer(ErrorMessageUtil msgUtil, FF4j ff4j) {
-        this.msgUtil = msgUtil;
-        this.ff4j = ff4j;
+        super(msgUtil, ff4j);
     }
 
     @Override
     public void review(CertifiedProductSearchDetails listing) {
         if (ff4j.check(FeatureList.EFFECTIVE_RULE_DATE)) {
             checkForInvalidCriteriaCombination(listing, criteriaB6Id, criteriaB10Id);
             checkForInvalidCriteriaCombination(listing, criteriaG8Id, criteriaG10Id);
+
+            initializeOldAndNewCriteria();
+            Iterator<Integer> oldCriteriaIdsIter = oldCriteriaIds.iterator();
+            Iterator<Integer> newCriteriaIdsIter = newCriteriaIds.iterator();
+            while (oldCriteriaIdsIter.hasNext() && newCriteriaIdsIter.hasNext()) {
+                checkForInvalidCriteriaCombination(listing, oldCriteriaIdsIter.next(), newCriteriaIdsIter.next());
+            }
         }
     }
 
     private void checkForInvalidCriteriaCombination(CertifiedProductSearchDetails listing, Integer criteriaIdA,
             Integer criteriaIdB) {
-
-        Optional<CertificationResult> certResultA = findCerificationResult(listing, criteriaIdA);
-        Optional<CertificationResult> certResultB = findCerificationResult(listing, criteriaIdB);
+        Optional<CertificationResult> certResultA = findCertificationResult(listing, criteriaIdA);
+        Optional<CertificationResult> certResultB = findCertificationResult(listing, criteriaIdB);
 
         if (certResultA.isPresent() && certResultB.isPresent()) {
+            final CertificationCriterion critA = certResultA.get().getCriterion();
+            final CertificationCriterion critB = certResultB.get().getCriterion();
             listing.getErrorMessages()
                     .add(msgUtil.getMessage("listing.criteria.invalidCombination",
-                            certResultA.get().getCriterion().getNumber(),
-                            certResultB.get().getCriterion().getNumber()));
+                            getTitleForErrorMessage(critA), getTitleForErrorMessage(critB)));
         }
     }
 
-    private Optional<CertificationResult> findCerificationResult(CertifiedProductSearchDetails listing,
+    private Optional<CertificationResult> findCertificationResult(CertifiedProductSearchDetails listing,
             Integer criteriaId) {
         return listing.getCertificationResults().stream()
                 .filter(cr -> cr.getCriterion().getId().equals(Long.valueOf(criteriaId)) && cr.isSuccess())
                 .findFirst();
     }
+
+    private String getTitleForErrorMessage(CertificationCriterion criterion) {
+        return criterion.getTitle().contains(CURES_UPDATE_IN_TITLE)
```

Comment:
I had added a method to the CertificaitonCriterion class called formatCriteriaNumber() wanting to keep the "(Cures Update)" formatting logic in one place. Could that be used here?